Assets/MixedRealityToolkit/Utilities/GameObjectManagement/IGameObjectCreatorHandler.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.GameObjectManagement
+{
+    /// <summary>
+    /// Optional interface that GameObjects (instantiated and recycled by a
+    /// GameObjectPool) can implement in order to handle preparation for
+    /// recycling and reuse.
+    /// </summary>
+    public interface IGameObjectCreatorHandler
```

Comment:
Do we need `Handler` suffix?

Reply:
What about `IGameObjectPoolEntity` or `Item` or something along those lines?

I think the handler suffix might confuse people with its usage as a handler to a service.

Reply:
Yes, or at least, something similar. The interface here is not an interface for a GameObject creator, but rather handles actions taken by the creator on GameObjects, such as when it is preparing to recycle a GameObject or is preparing to use it.

Reply:
In this way a GameObject that is in the pool can have a script that implements that interface (if needed) to carry out actions when recycling or use preparations are carried out on it. Perhaps "**IGameObjectCreatorListener**" or similar would be better as a name?

Reply:
Updated the name to IGameObjectCreatorListener